### PR TITLE
Feature/virtual backgrounds

### PR DESCRIPTION
**Pull request by dmtrhfr**
Initial PR to discuss some implementations for the addition of Virtualbackgrounds to the ansible-role.

Documentation not yet updated.

### CODE REVIEW

**Comment by d3hof** [comment]
Currently i am adding the virtual backgrounds to the hmlt5-cofnig via blockinfile, and i have the issue with an resulting blank line, because of jinja2 templating.

Maybe it's better to write it directly into the bbb_meteor variable.

I added two different options to add the virtual-backgrounds, either by download or from the files-folder in the ansible project above the role. Not sure if that's needed, but i used the first one in one of our plays and think it's a neat option. Currently only a single url is supported - file names are defined by the bbb_virtual_backgrounds list.

**Comment by dmtrhfr** [comment]
Will fix the ansible linter issues asap

**Comment by BrutalBirdie** [comment]
the linter needs a new config anyway, It should not be an error if the first char is not BIG.

**Comment by dmtrhfr** [comment]
I am not really happy with this part - it adds comments and an empty new line into the bbb-html5.yaml.
Any ideas to do this? Or should i remove this part and define everything in bbb_meteor. What do you think?

```
- name: add config for vb to html config
  become: true
  blockinfile:
    path: /etc/bigbluebutton/bbb-html5.yml
    marker_begin: "begin virtual backgrounds"
    marker_end: "end virtual background"
    insertafter: "public:"
    block: |2
        virtualBackgrounds:
          fileNames:
          {% for item in bbb_virtual_backgrounds -%}
            - "{{ item }}"
          {% endfor -%}
  notify:
    - restart bbb-html5
```

**Comment by einhirn** [comment]
I actually went the other way round: Adding the files to "bbb_meteor" variable and just copying them over to the right place. Of course I was lazy and didn't automate the thumbnail generation...

```
- name: "Copy meteor background images"
  copy:
    src: "backgrounds/{{ item }}"
    dest: "/usr/share/meteor/bundle/programs/web.browser/app/resources/images/virtual-backgrounds/{{ item }}"
    mode: 0444
    owner: root
    group: root
  with_items: "{{ bbb_meteor.public.virtualBackgrounds.fileNames }}"
  tags: bg-image

- name: "Copy meteor background image thumbnails"
  copy:
    src: "backgrounds/thumbnails/{{ item }}"
    dest: "/usr/share/meteor/bundle/programs/web.browser/app/resources/images/virtual-backgrounds/thumbnails/{{ item }}"
    mode: 0444
    owner: root
    group: root
  with_items: "{{ bbb_meteor.public.virtualBackgrounds.fileNames }}"
  tags: bg-image

**Comment by dmtrhfr** [comment]
That's a good suggestion. Trying to implement it with my Download feature.

**Comment by dmtrhfr** [comment]
Does anybody know what happens if you define the virtual backgrounds in bbb-html5.yml before they are available in the directory?

**Comment by einhirn** [comment]
I guess nothing special except broken media in Web Session...

**Comment by dmtrhfr** [comment]
Is there a way to only run the play if the virtualBackgrounds section is present in the html5 config?

**Comment by dmtrhfr** [comment]
Should be finished now, couldn't test yet.

File: tasks/virtual-backgrounds.yml [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+- name: Copy shell script for creation of thumbnails
+  copy:
+    src: vb-thumbnail.sh
+    dest: /root/vb-thumbnail.sh
+    mode: 0700
+    owner: root
+    group: root
+
+- name: Create thumbnails
+  command: '/root/vb-thumbnail.sh'
```
[the remark this answers is not in the document]
Comment (defnull) [reply tasks/virtual-backgrounds.yml:44]: Another option would be `ansible.builtin.shell`. Uploading, running and then removing a script every time may be unnecessary.

[the remark this answers is not in the document]
Comment (dmtrhfr) [reply tasks/virtual-backgrounds.yml:44]: wdym with calling convert directly?

[the remark this answers is not in the document]
Comment (dmtrhfr) [reply tasks/virtual-backgrounds.yml:44]: Ok you mean just call the shell script directly without copying the file. Ok i can do that, seems reasonable to me. thanks for the feedback.

[the remark this answers is not in the document]
Comment (dmtrhfr) [reply tasks/virtual-backgrounds.yml]: So i moved it all to a single command, and moved the restart out of the task into the parent task.

[the remark this answers is not in the document]
Comment (dmtrhfr) [reply tasks/virtual-backgrounds.yml]: What i didn't accomplish yet is to not run the converting if that already happened before.

[the remark this answers is not in the document]
Comment (dmtrhfr) [reply tasks/virtual-backgrounds.yml]: Another option would be to move that out of the role and you have to provide the thumbnails from the files.

**Pull request closed without merging**